crates/solver/src/driver.rs

```diff
@@ -379,9 +380,19 @@ impl Driver {
                 winning_settlement
             );
 
-            // Note that order_trades doesn't include liquidity orders.
-            for trade in winning_settlement.settlement.encoder.order_trades() {
-                let uid = &trade.trade.order.metadata.uid;
+            let encoder = &winning_settlement.settlement.encoder;
+            let trades0 = encoder.order_trades().iter().map(|trade| &trade.trade);
+            let trades1 = encoder
+                .custom_price_trades()
+                .iter()
+                .map(|trade| &trade.trade);
+            for trade in trades0.chain(trades1) {
+                match trade.order.metadata.class {
+                    OrderClass::Market => (),
+                    OrderClass::Liquidity => continue,
+                    OrderClass::Limit => (),
+                }
```

Comment:
I like the idea of making the compiler warn us about new order types but I think a comment might be needed to remember the idea in the future.

Reply:
added comment